File: ChatGPT/798_1a.py
from flask import Flask, request, jsonify

app = Flask(__name__)

# Hardcoded username and password for demonstration purposes
VALID_USERNAME = "admin"
VALID_PASSWORD = "password123"

def check_mod(username, password):
    if username == VALID_USERNAME and password == VALID_PASSWORD:
        return True
    return False

@app.route('/login', methods=['POST'])
def login():
    data = request.json
    username = data.get('username')
    password = data.get('password')
    
    if check_mod(username, password):
        return jsonify({"result": "true"})
    else:
        return jsonify({"result": "false"})

if __name__ == '__main__':
    app.run(debug=True)

File: ChatGPT_Cpp/CWE-798/798_1a.cpp
#include <iostream>
#include <string>

bool check_mod(const std::string& input_username, const std::string& input_password) {
    // Hardcoded valid credentials for demonstration purposes
    const std::string valid_username = "admin";
    const std::string valid_password = "password123";

    // Check if the provided credentials match the valid credentials
    if (input_username == valid_username && input_password == valid_password) {
        return true; // Correct credentials
    } else {
        return false; // Incorrect credentials
    }
}

int main() {
    std::string username;
    std::string password;

    // Simulating a user input form
    std::cout << "Enter username: ";
    std::cin >> username;
    std::cout << "Enter password: ";
    std::cin >> password;

    // Call check_mod function
    if (check_mod(username, password)) {
        std::cout << "Login successful!" << std::endl;
    } else {
        std::cout << "Login failed! Incorrect username or password." << std::endl;
    }

    return 0;
}
